src/pudl_rmi/coordinate.py

```diff
@@ -0,0 +1,247 @@
+        regenerate any or all of the interconnected RMI outputs. Each `get_`
+        method in this object grabs a pickle file off disk if it is available,
+        or generates the output if it is not available and/or if you set
+        clobber to True.
+
+        Because some outputs rely on others, this object enables clobbering of
+        both the main output and the outputs which the main object relies on.
+        This in effect enables a clobber and deep clobbers.
+
+        Most of the outputs are generated via a ``execute()`` function from the
+        cooresponding module for that output. The plant-part list is generated
+        from the `pudl_out` object.
+
+        Args:
+            pudl_out (object): instance of `pudl.output.pudltabl.PudlTabl()`.
+                The frequency (`freq`) of `pudl_out` must be `AS`. For best
+                results  `fill_fuel_cost`, `roll_fuel_cost`, and `fill_net_gen`
+                should all be True.
+        """
+        self.pudl_out = pudl_out
+        if pudl_out.freq != 'AS':
+            raise AssertionError(
+                f"Frequency of `pudl_out` must be `AS` but got {pudl_out.freq}"
+            )
+
+    def get_plant_part_list(self, clobber=False):
+        """
+        Get the master unit list; generate it or get if from a file.
+
+        If you generate the MUL, it will be saved at the file path given. The
+        plant-parts list is generated via the pudl_out object.
+
+        Args:
+            clobber (boolean): True if you want to regenerate the master unit
+                list whether or not the output is already pickled. Default is
+                False.
+        """
+        if not pudl_rmi.FILE_PATH_PLANT_PARTS_EIA.is_file() or clobber:
+            logger.info(
+                "Master unit list not found "
+                f"{pudl_rmi.FILE_PATH_PLANT_PARTS_EIA} Generating a new master"
+                " unit list. This should take ~10 minutes."
+            )
+            # actually make the master plant parts list
+            plant_parts_eia = self.pudl_out.plant_parts_eia()
+            # export
+            plant_parts_eia.to_pickle(
+                pudl_rmi.FILE_PATH_PLANT_PARTS_EIA, compression='gzip'
+            )
+        else:
+            logger.info(
+                "Reading the plant part list from "
+                f"{pudl_rmi.FILE_PATH_PLANT_PARTS_EIA}"
+            )
+            plant_parts_eia = pd.read_pickle(
+                pudl_rmi.FILE_PATH_PLANT_PARTS_EIA, compression='gzip'
+            )
+        return plant_parts_eia
+
+    def get_deprish(self, clobber=False):
+        """
+        Generate of grab the cleaned deprecaition studies.
+
+        Args:
+            clobber (boolean): True if you want to regenerate the depreciation
+                data whether or not the output is already pickled. Default is
+                False.
+        """
+        if not pudl_rmi.FILE_PATH_DEPRISH.is_file() or clobber:
```

Comment:
If that path does exist but isn't a file, I think you'll get undesirable behavior here. (it could be a directory, or a symlink, etc.)

Reply:
[`is_file()`](https://docs.python.org/3/library/pathlib.html#pathlib.Path.is_file) will not return True if it is a directory.

Reply:
If it's a directory `is_file()` will return `False` and it'll try to create the file. But then what happens? Does it put the file inside the directory? Does it clobber the directory and everything inside it? Does it give an error because it's trying to write out to a path that already exists even though it's not a file?